omnipaxos/tests/integration_test.rs

```diff
@@ -0,0 +1,274 @@
+// To run: cargo test --test integration_test --features "toml_config macros"
+#![cfg(feature = "macros")]
+#![cfg(feature = "toml_config")]
+use commitlog::LogOptions;
+use omnipaxos::macros::Entry;
+use omnipaxos::messages::Message;
+#[cfg(not(feature = "macros"))]
+use omnipaxos::storage::Entry;
+use omnipaxos::storage::Snapshot;
+use omnipaxos::util::LogEntry;
+use omnipaxos::{ClusterConfig, OmniPaxos, OmniPaxosConfig, ServerConfig};
+use omnipaxos_storage::persistent_storage::{PersistentStorage, PersistentStorageConfig};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+// Original: https://omnipaxos.com/docs/omnipaxos/omnipaxos/
+#[cfg_attr(feature = "macros", derive(Entry))]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct KeyValue {
+    pub key: String,
+    pub value: u64,
+}
+
+#[cfg(not(feature = "macros"))]
+impl Entry for KeyValue {
+    type Snapshot = KVSnapshot;
+}
+
+// Original: https://omnipaxos.com/docs/omnipaxos/compaction/
+#[derive(Clone, Debug)]
+pub struct KVSnapshot {
+    snapshotted: HashMap<String, u64>,
+}
+
+impl Snapshot<KeyValue> for KVSnapshot {
+    fn create(entries: &[KeyValue]) -> Self {
+        let mut snapshotted = HashMap::new();
+        for e in entries {
+            let KeyValue { key, value } = e;
+            snapshotted.insert(key.clone(), *value);
+        }
+        Self { snapshotted }
+    }
+
+    fn merge(&mut self, delta: Self) {
+        for (k, v) in delta.snapshotted {
+            self.snapshotted.insert(k, v);
+        }
+    }
+
+    fn use_snapshots() -> bool {
+        true
+    }
+}
+
+// Original: https://omnipaxos.com/docs/omnipaxos/omnipaxos/
+#[test]
+fn omnipaxos() {
+    // Use case of creating a node
+    // https://omnipaxos.com/docs/omnipaxos/omnipaxos/
+    use omnipaxos_storage::memory_storage::MemoryStorage;
+
+    // Original: https://omnipaxos.com/docs/omnipaxos/omnipaxos/
+    // configuration with id 1 and a cluster with 3 nodes
+    let cluster_config = ClusterConfig {
+        configuration_id: 1,
+        nodes: vec![1, 2, 3],
+        ..Default::default()
+    };
+
+    // create the replica 2 in this cluster (other replica instances are created similarly with pid 1 and 3 on the other nodes)
+    let server_config = ServerConfig {
+        pid: 2,
+        ..Default::default()
+    };
+
+    // Combined OmniPaxos config with both clsuter-wide and server specific configurations
+    let omnipaxos_config = OmniPaxosConfig {
+        cluster_config,
+        server_config,
+    };
+
+    let storage = MemoryStorage::default();
+    let mut omni_paxos: OmniPaxos<KeyValue, MemoryStorage<KeyValue>> =
+        omnipaxos_config.build(storage).unwrap();
+
+    // Original: https://omnipaxos.com/docs/omnipaxos/omnipaxos/
+    // Use case of creating a node with TOML config
+    let config_file_path = "tests/config/node1.toml";
+    let omnipaxos_config = OmniPaxosConfig::with_toml(config_file_path).unwrap();
```

Comment:
I think we should have each code example from the website as their own function with a link to the corresponding docs file and section.

In this way, if a function is changed or does not compile, we know exactly where in the docs to update. 

So here it could be like this
```
// This corresponds to the example:
// https://github.com/haraldng/omnipaxos/blob/master/docs/omnipaxos/index.md#creating-a-node
fn creating_a_node() {
     ...
}
```